### PR TITLE
TG-2387 Implemented tax and basic sizing

### PR DESCRIPTION
**Pull request by paulkennethkent**
Tax is added to all products except when the object size and category match the exempt criteria. Sizing implementation can be improved but I think this should be considered when we work on the stock control feature which will require big changes to the models.

### CODE REVIEW

File: src/main/resources/templates/ProductPage.html [hunk 1]
```diff
@@ -13,6 +13,11 @@
                 <div th:text="${product.price}"/>
                 <form th:action="@{/addToCart}" th:object="${product}" method="post">
                     <input type="hidden" th:field="*{id}"/>
+                      <select name="size" required="required">
+                        <option value="">-- Please select a size --</option>  
+                        <option value="Child">Child</option>
+                        <option value="Adults">Adult</option>
```
Comment (JohnLBergqvist) [inline src/main/resources/templates/ProductPage.html]: The value here should be `Adult`, no?

**Review by JohnLBergqvist — CHANGES_REQUESTED** [review]

**Review by JohnLBergqvist — APPROVED** [review]

**Review by jgwilson42 — CHANGES_REQUESTED** [review]
Please could you replace BigDecimal with double?

File: src/main/java/com/diffblue/demo/ecommerce/models/Order.java [hunk 2]
```diff
@@ -57,7 +57,7 @@
    * @param phoneNumber - customer phone number
    * @param shippingAddress - shipping address
    */
-  public Order(String orderRef, String firstName, String lastName,
+  public Order(String orderRef, String firstName, String lastName,
```
Comment (peterschrammel) [inline src/main/java/com/diffblue/demo/ecommerce/models/Order.java]: trailing whitespace

File: src/main/java/com/diffblue/demo/ecommerce/models/OrderItem.java [hunk 3]
```diff
@@ -1,6 +1,6 @@
 package com.diffblue.demo.ecommerce.models;
 
-// Copyright 2016-2017 DiffBlue limited. All rights reserved.
+// Copyright 2016-2018 DiffBlue limited. All rights reserved.
```
Comment (peterschrammel) [inline src/main/java/com/diffblue/demo/ecommerce/models/OrderItem.java]: Diffblue (all files)

File: src/main/java/com/diffblue/demo/ecommerce/models/Cart.java [hunk 4]
```diff
@@ -6,11 +6,16 @@
 import java.util.HashMap;
 import java.util.Map;
 
+
```
Comment (peterschrammel) [inline src/main/java/com/diffblue/demo/ecommerce/models/Cart.java]: spurious newlines

File: src/main/java/com/diffblue/demo/ecommerce/models/Cart.java [hunk 5]
```diff
@@ -6,11 +6,16 @@
...
+
 public class Cart {
 
   private Map<Product, Integer> cartItems;
   private BigDecimal subtotal;
   private BigDecimal shipping;
+  private BigDecimal tax;
```
Comment (peterschrammel) [inline src/main/java/com/diffblue/demo/ecommerce/models/Cart.java]: Shouldn't this be double for now?

File: src/main/java/com/diffblue/demo/ecommerce/models/Cart.java [hunk 6]
```diff
@@ -6,8 +6,6 @@
 import java.util.HashMap;
 import java.util.Map;
 
-
-
```
Comment (peterschrammel) [inline src/main/java/com/diffblue/demo/ecommerce/models/Cart.java]: In a proper dev workflow these fixes should be squashed into the commit that created the problem in the first place (if this commit is in the same PR).

File: src/main/resources/templates/ProductPage.html [hunk 7]
```diff
@@ -16,7 +16,7 @@
                       <select name="size" required="required">
                         <option value="">-- Please select a size --</option>  
                         <option value="Child">Child</option>
-                        <option value="Adults">Adult</option>
+                        <option value="Adult">Adult</option>
```
Comment (peterschrammel) [inline src/main/resources/templates/ProductPage.html]: squash this into the original commit

**Review by peterschrammel — CHANGES_REQUESTED** [review]
Commit 
"Merge branch 'master' into cart-tax" should be removed by rebasing the branch on master.

**Comment by peterschrammel** [comment]
Note: In a proper dev workflow these fixes should be squashed into the commit that created the problem in the first place (if this commit is in the same PR). I'm reviewing PRs commit by commit - if something is fixed up in a later commit I will still have put a comment on the earlier commit which then appears as 'outdated'.

**Review by peterschrammel — APPROVED** [review]
LGTM

**Pull request merged**